### ISSUE TITLE
java.lang.RuntimeException: Unable to create application <>: java.lang.RuntimeException: Package manager has died

### ISSUE THREAD

**Issue by jaredsburrows** [bitstadium/HockeySDK-Android#214]
## Hockey app:

4.1.2
## Android:

6.0.1
## Manufacturer:

samsung
## Model:

SM-G935F
## Exception:

```
java.lang.RuntimeException: Unable to create application <>: java.lang.RuntimeException: Package manager has died
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6324)
	at android.app.ActivityThread.access$1800(ActivityThread.java:222)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1861)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:158)
	at android.app.ActivityThread.main(ActivityThread.java:7229)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by: java.lang.RuntimeException: Package manager has died
	at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:196)
	at net.hockeyapp.android.metrics.TelemetryContext.configApplicationContext(TelemetryContext.java:168)
	at net.hockeyapp.android.metrics.TelemetryContext.<init>(TelemetryContext.java:119)
	at net.hockeyapp.android.metrics.MetricsManager.register(MetricsManager.java:225)
	at net.hockeyapp.android.metrics.MetricsManager.register(MetricsManager.java:170)
	at net.hockeyapp.android.metrics.MetricsManager.register(MetricsManager.java:158)
	at <>
	at <>
	at com.microsoft.intune.mam.client.app.MAMApplication$Impl.onCreate(MAMApplication.java:98)
	at com.microsoft.intune.mam.client.app.MAMApplication.onCreate(MAMApplication.java:48)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1036)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6321)
	... 8 more
Caused by: android.os.DeadObjectException
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(Binder.java:503)
	at android.content.pm.IPackageManager$Stub$Proxy.getPackageInfo(IPackageManager.java:2721)
	at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:191)
	... 19 more
```

**Comment by jaredsburrows** [comment]
My suggestion is to go ahead and catch with `Exception` instead of `PackageManager.NameNotFoundException`, here: https://github.com/bitstadium/HockeySDK-Android/blob/ce656a828dc562dcf7db2dfd6de9032e1cef06e9/hockeysdk/src/main/java/net/hockeyapp/android/metrics/TelemetryContext.java.

This way, the `java.lang.RuntimeException: Package manager has died` should be handled gracefully instead of a crash. Then the `finally` should be called after.

**Comment by TroubleMakerBen** [comment]
Hey @jaredsburrows 
Apologies for issues this causes and thanks for the suggestions. They make a lot of sense to me. We're looking for a fix soon, but I can't give you an ETA for an SDK update.

Best,
Benjamin

**Comment by jaredsburrows** [comment]
@TroubleMakerBen For this, I can submit a quick PR to use `Exception` instead of `PackageManager.NameNotFoundException`.

**Comment by TroubleMakerBen** [comment]
The fix will be part of 4.1.4, but I can't give you an ETA. I'll keep this issue open until the fix has been released.

**Comment by jaredsburrows** [comment]
@TroubleMakerBen Thanks so much! Closing this issue now.
